### PR TITLE
libraries vs frameworks article

### LINKED ISSUE #7840: Understand the differences between a library and a framework [6/5/2022]
* Proposed title: Understand the differences between a library and a framework
* Proposed by: johanneq
* Related areas: None of the above
* Content type: Explanation (https://documentation.divio.com/explanation/)
* Deadline: 6/5/2022
* Deadline explanation: One month will be fine - the pathways on DGC where this content belongs will be live for ETT prior to this date, but they will not need to review right away.
* [Draft](https://docs.google.com/document/d/13LZou1HXSxAnpI5akviuXI14pVJqGo3k6lN10VNyNe4/edit?usp=sharing)
* Content summary: This article is part of the Web Fundamentals Course launching to the public in September, and to the ETT program in April. In this article, student's learn about scope and how it works in JavaScript.

### PR DESCRIPTION
**Pull request by rachelandrew**
Fixes #7840 

Another of the pieces of content we are hosting for the ETT course.

Preview: https://deploy-preview-7975--web-dev-staging.netlify.app/js-libraries-vs-frameworks/

### CODE REVIEW

File: src/site/content/en/web-dev-basics-one/js-libraries-vs-frameworks/index.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
+---
+layout: post
+title: The difference between JavaScript libraries and frameworks
+description: Understand the differences between frameworks and libraries in the context of a client-side JavaScript environment.
```
Comment (umaar) [inline src/site/content/en/web-dev-basics-one/js-libraries-vs-frameworks/index.md:4]:
```suggestion
description: Understand the differences between frameworks and libraries in the context of a client-side JavaScript environment. 
authors:
  - umarhansa
```

**Review by umaar — COMMENTED** [review]
A few suggestions!

File: src/site/content/en/web-dev-basics-one/js-libraries-vs-frameworks/index.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+### Framework
+
+Frameworks tend to be bigger than libraries and contribute more to overall page weight. In fact, a framework can include a library. 
+
+This example shows a plain framework without a library and uses [Vue](https://vuejs.org/), which is a [popular](https://2021.stateofjs.com/en-US/libraries/front-end-frameworks/#front_end_frameworks_experience_ranking) JavaScript framework:
+
+```html
+<!-- index.html -->
+<div id="main">
+  {{ message }}
```
Comment (umaar) [inline src/site/content/en/web-dev-basics-one/js-libraries-vs-frameworks/index.md:60]:
```suggestion
  {{ '{{ message }}' }}
```

Looks like the web.dev templating system, via eleventy, was interpolating the double mustache tags which broke the code sample. I've cloned this branch locally, have run the build, and can confirm we can escape it with this syntax `{{ '{{ message }}' }}` which fixes the issue. Screenshot confirmation of the rendered page below:

image

Happy to use another technique though.

**Review by umaar — COMMENTED** [review]
One last issue (from me) with a template interpolation issue which invalidates a code sample. Apart from that, LGTM!

**Comment by rachelandrew** [comment]
@umaar can you take a look at this one now?

**Comment by umaar** [comment]
Nice, can see it fixed on the preview URL. LGTM!

**Pull request merged**